--- /net/minecraft/entity/ai/attributes
+++ /net/minecraft/entity/ai/attributes
@@ -79,7 +79,7 @@
         if(this.getModifier(var1.getID()) != null) {
             throw new IllegalArgumentException("Modifier is already applied on this attribute!");
         } else {
-            Object var2 = (Set)this.mapByName.get(var1.getName());
+            Set<AttributeModifier> var2 = (Set)this.mapByName.get(var1.getName());
             if(var2 == null) {
                 var2 = Sets.newHashSet();
                 this.mapByName.put(var1.getName(), var2);
@@ -127,7 +127,7 @@
         side = Side.CLIENT
     )
     public void removeAllModifiers() {
-        Collection var1 = this.getModifiers();
+        Collection<AttributeModifier> var1 = this.getModifiers();
         if(var1 != null) {
             for(AttributeModifier var3 : Lists.newArrayList(var1)) {
                 this.removeModifier(var3);
