--- /net/minecraft/util
+++ /net/minecraft/util
@@ -20,9 +20,9 @@
 
     public IntIdentityHashBiMap(int var1) {
         var1 = (int)((float)var1 / 0.8F);
-        this.values = (Object[])(new Object[var1]);
+        this.values = (K[])(new Object[var1]);
         this.intKeys = new int[var1];
-        this.byId = (Object[])(new Object[var1]);
+        this.byId = (K[])(new Object[var1]);
     }
 
     public int getId(K var1) {
@@ -55,15 +55,15 @@
     private void grow(int var1) {
         Object[] var2 = this.values;
         int[] var3 = this.intKeys;
-        this.values = (Object[])(new Object[var1]);
+        this.values = (K[])(new Object[var1]);
         this.intKeys = new int[var1];
-        this.byId = (Object[])(new Object[var1]);
+        this.byId = (K[])(new Object[var1]);
         this.nextFreeIndex = 0;
         this.mapSize = 0;
 
         for(int var4 = 0; var4 < var2.length; ++var4) {
             if(var2[var4] != null) {
-                this.put(var2[var4], var3[var4]);
+                this.put((K) var2[var4], var3[var4]);
             }
         }
 
