--- /net/minecraft/client/resources
+++ /net/minecraft/client/resources
@@ -33,7 +33,7 @@
         HashSet var1 = Sets.newHashSet();
         File var2 = new File(this.resourcePackFile, "assets/");
         if(var2.isDirectory()) {
-            for(File var6 : var2.listFiles(DirectoryFileFilter.DIRECTORY)) {
+            for(File var6 : var2.listFiles((java.io.FileFilter)DirectoryFileFilter.DIRECTORY)) {
                 String var7 = getRelativeName(var2, var6);
                 if(var7.equals(var7.toLowerCase())) {
                     var1.add(var7.substring(0, var7.length() - 1));
