--- /net/minecraft/entity/player
+++ /net/minecraft/entity/player
@@ -1255,7 +1255,7 @@
     }
 
     private boolean isInBed() {
-        return this.worldObj.getBlockState(this.playerLocation).getBlock() == Blocks.BED;
+        return xyz.openmodloader.event.impl.PlayerEvent.SleepCheck.handle(this, playerLocation);
     }
 
     @Nullable
@@ -1653,8 +1653,8 @@
             this.playEquipSound(var2);
             this.inventory.offHandInventory[0] = var2;
         } else if(var1.getSlotType() == EntityEquipmentSlot.a.ARMOR) {
-            this.playEquipSound(var2);
-            this.inventory.armorInventory[var1.getIndex()] = var2;
+            xyz.openmodloader.event.impl.ArmorEvent event = xyz.openmodloader.event.impl.ArmorEvent.handle(this, var2, var1);
+            if (event != null) { this.inventory.armorInventory[event.getSlot().getIndex()] = event.getArmor(); this.playEquipSound(var2); } else {inventory.addItemStackToInventory(var2); return;}
         }
 
     }
