--- /net/minecraft/tileentity
+++ /net/minecraft/tileentity
@@ -245,6 +245,7 @@
     }
 
     public static int getItemBurnTime(ItemStack var0) {
+        if(true) return xyz.openmodloader.registry.FuelRegistry.getItemBurnTime(var0);
         if(var0 == null) {
             return 0;
         } else {
@@ -269,6 +270,7 @@
     }
 
     public static boolean isItemFuel(ItemStack var0) {
+        if (true) return xyz.openmodloader.registry.FuelRegistry.isItemFuel(var0);
         return getItemBurnTime(var0) > 0;
     }
 
