--- /net/minecraft/entity
+++ /net/minecraft/entity
@@ -205,6 +205,7 @@
         this.dataManager.register(CUSTOM_NAME, "");
         this.dataManager.register(SILENT, Boolean.valueOf(false));
         this.dataManager.register(field_189655_aD, Boolean.valueOf(false));
+        xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.EntityEvent.Constructing(this));
         this.entityInit();
     }
 
@@ -325,6 +326,7 @@
     }
 
     public void onUpdate() {
+        xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.UpdateEvent.EntityUpdate(this));
         if(!this.worldObj.isRemote) {
             this.setFlag(6, this.isGlowing());
         }
@@ -1526,6 +1528,7 @@
 
     public boolean startRiding(Entity var1, boolean var2) {
         if(var2 || this.canBeRidden(var1) && var1.canFitPassenger(this)) {
+            if (!xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.EntityEvent.Mount(this, var1))) return false;
             if(this.isRiding()) {
                 this.dismountRidingEntity();
             }
@@ -1551,6 +1554,7 @@
 
     public void dismountRidingEntity() {
         if(this.ridingEntity != null) {
+            if (!xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.EntityEvent.Unmount(this, ridingEntity))) return;
             Entity var1 = this.ridingEntity;
             this.ridingEntity = null;
             var1.removePassenger(this);
@@ -1771,6 +1775,7 @@
     }
 
     public void onStruckByLightning(EntityLightningBolt var1) {
+        if (!xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.EntityEvent.LightningStruck(this, var1))) return;
         this.attackEntityFrom(DamageSource.lightningBolt, 5.0F);
         ++this.fire;
         if(this.fire == 0) {
@@ -1908,6 +1913,7 @@
             this.worldObj.theProfiler.startSection("changeDimension");
             MinecraftServer var2 = this.getServer();
             int var3 = this.dimension;
+            var1 = xyz.openmodloader.event.impl.EntityEvent.ChangeDimension.handle(this, var3, var1);
             WorldServer var4 = var2.worldServerForDimension(var3);
             WorldServer var5 = var2.worldServerForDimension(var1);
             this.dimension = var1;
@@ -2228,9 +2234,11 @@
     }
 
     public void addTrackingPlayer(EntityPlayerMP var1) {
+        xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.PlayerEvent.Track.Start(var1, this));
     }
 
     public void removeTrackingPlayer(EntityPlayerMP var1) {
+        xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.PlayerEvent.Track.Stop(var1, this));
     }
 
     public float getRotatedYaw(Rotation var1) {
