--- /net/minecraft/client/network
+++ /net/minecraft/client/network
@@ -669,10 +669,12 @@
 
     public void handleChat(SPacketChat var1) {
         PacketThreadUtil.checkThreadAndEnqueue(var1, this, this.gameController);
+        ITextComponent component = xyz.openmodloader.event.impl.MessageEvent.Chat.handle(var1.getChatComponent(), xyz.openmodloader.launcher.strippable.Side.CLIENT);
+        if (component == null) return;
         if(var1.getType() == 2) {
-            this.gameController.ingameGUI.setRecordPlaying(var1.getChatComponent(), false);
+            this.gameController.ingameGUI.setRecordPlaying(component, false);
         } else {
-            this.gameController.ingameGUI.getChatGUI().printChatMessage(var1.getChatComponent());
+            this.gameController.ingameGUI.getChatGUI().printChatMessage(component);
         }
 
     }
