--- /net/minecraft/inventory
+++ /net/minecraft/inventory
@@ -11,17 +11,13 @@
 import net.minecraft.init.Blocks;
 import net.minecraft.init.Items;
 import net.minecraft.init.SoundEvents;
-import net.minecraft.inventory.Container;
-import net.minecraft.inventory.IContainerListener;
-import net.minecraft.inventory.IInventory;
-import net.minecraft.inventory.InventoryBasic;
-import net.minecraft.inventory.Slot;
 import net.minecraft.item.EnumDyeColor;
 import net.minecraft.item.ItemStack;
 import net.minecraft.stats.StatList;
 import net.minecraft.util.SoundCategory;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
+import xyz.openmodloader.event.impl.EnchantmentEvent;
 
 public class ContainerEnchantment extends Container {
    public IInventory tableInventory;
@@ -209,6 +205,7 @@
          if(!this.worldPointer.isRemote) {
             List var6 = this.getEnchantmentList(var3, var2, this.enchantLevels[var2]);
             boolean var7 = var3.getItem() == Items.BOOK;
+            var6 = EnchantmentEvent.Item.handle(var1, var3, var4, var5, var6);
             if(var6 != null) {
                var1.removeExperienceLevel(var5);
                if(var7) {
