--- /net/minecraft/block/state
+++ /net/minecraft/block/state
@@ -309,6 +309,7 @@
         }
 
         public float getPlayerRelativeBlockHardness(EntityPlayer var1, World var2, BlockPos var3) {
+            if(true) return xyz.openmodloader.event.impl.BlockEvent.DigSpeed.handle(var1.isCreative() ? 1.0F : this.block.getPlayerRelativeBlockHardness(this, var1, var2, var3), var2, var2.getBlockState(var3), var3);
             return this.block.getPlayerRelativeBlockHardness(this, var1, var2, var3);
         }
 
