--- /net/minecraft/entity/item
+++ /net/minecraft/entity/item
@@ -262,6 +262,7 @@
 
     public void onCollideWithPlayer(EntityPlayer var1) {
         if(!this.worldObj.isRemote) {
+            if (!xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.EntityEvent.ItemPickup(var1, this))) return;
             ItemStack var2 = this.getEntityItem();
             int var3 = var2.stackSize;
             if(this.delayBeforeCanPickup == 0 && (this.owner == null || 6000 - this.age <= 200 || this.owner.equals(var1.getName())) && var1.inventory.addItemStackToInventory(var2)) {
