--- /net/minecraft/client/renderer
+++ /net/minecraft/client/renderer
@@ -897,6 +897,7 @@
 
                 try {
                     this.mc.currentScreen.drawScreen(var22, var23, var1);
+                    xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.GuiEvent.Draw(mc.currentScreen, var1));
                 } catch (Throwable var16) {
                     CrashReport var26 = CrashReport.makeCrashReport(var16, "Rendering screen");
                     CrashReportCategory var13 = var26.makeCategory("Screen render details");
