--- /net/minecraft/world
+++ /net/minecraft/world
@@ -846,6 +846,7 @@
                 this.updateAllPlayersSleepingFlag();
             }
 
+            if (!xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.EntityEvent.Join(var1)) && !var4) return false;
             this.getChunkFromChunkCoords(var2, var3).addEntity(var1);
             this.loadedEntityList.add(var1);
             this.onEntityAdded(var1);
@@ -1708,7 +1709,8 @@
     }
 
     public Explosion newExplosion(@Nullable Entity var1, double var2, double var4, double var6, float var8, boolean var9, boolean var10) {
-        Explosion var11 = new Explosion(this, var1, var2, var4, var6, var8, var9, var10);
+        Explosion var11 = xyz.openmodloader.event.impl.ExplosionEvent.onExplosion(this, var1, var2, var4, var6, var8, var9, var10);
+        if(var11 == null) return null;
         var11.doExplosionA();
         var11.doExplosionB(true);
         return var11;
@@ -1883,6 +1885,7 @@
     }
 
     public void tick() {
+        xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.UpdateEvent.WorldUpdate(this));
         this.updateWeather();
     }
 
@@ -2320,9 +2323,12 @@
     }
 
     public void loadEntities(Collection<Entity> var1) {
+        if(false)
         this.loadedEntityList.addAll(var1);
 
         for(Entity var3 : var1) {
+            if (!xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.EntityEvent.Join(var3))) continue;
+            this.loadedEntityList.add(var3);
             this.onEntityAdded(var3);
         }
 
@@ -2590,6 +2596,7 @@
         }
 
         if(!this.loadedEntityList.contains(var1)) {
+            if (!xyz.openmodloader.OpenModLoader.INSTANCE.getEventBus().post(new xyz.openmodloader.event.impl.EntityEvent.Join(var1))) return;
             this.loadedEntityList.add(var1);
         }
 
