--- /net/minecraft/client/renderer
+++ /net/minecraft/client/renderer
@@ -223,7 +223,7 @@
         float var11 = MathHelper.sin(var6 * 3.1415927F);
         GlStateManager.rotate(var5 * var11 * 70.0F, 0.0F, 1.0F, 0.0F);
         GlStateManager.rotate(var5 * var10 * -20.0F, 0.0F, 0.0F, 1.0F);
-        EntityPlayerSP var12 = this.mc.thePlayer;
+        AbstractClientPlayer var12 = this.mc.thePlayer;
         this.mc.getTextureManager().bindTexture(var12.getLocationSkin());
         GlStateManager.translate(var5 * -1.0F, 3.6F, 3.5F);
         GlStateManager.rotate(var5 * 120.0F, 0.0F, 0.0F, 1.0F);
