--- /net/minecraft/client/renderer/entity
+++ /net/minecraft/client/renderer/entity
@@ -53,6 +53,7 @@
     }
 
     public void doRender(T var1, double var2, double var4, double var6, float var8, float var9) {
+        if(!xyz.openmodloader.event.impl.RenderEvent.Entities.onRender(this, var1, var2, var4, var6, var8, var9)) return;
         if(!this.renderOutlines) {
             this.renderName(var1, var2, var4, var6);
         }
@@ -271,11 +272,13 @@
                 double var10 = this.renderManager.getDistanceToCamera(var1.posX, var1.posY, var1.posZ);
                 float var12 = (float)((1.0D - var10 / 256.0D) * (double)this.shadowOpaque);
                 if(var12 > 0.0F) {
+                    if (!xyz.openmodloader.event.impl.RenderEvent.Shadows.onRender(var1, var2, var4, var6, var8, var9)) return;
                     this.renderShadow(var1, var2, var4, var6, var12, var9);
                 }
             }
 
             if(var1.canRenderOnFire() && (!(var1 instanceof EntityPlayer) || !((EntityPlayer)var1).isSpectator())) {
+                if (!xyz.openmodloader.event.impl.RenderEvent.Fires.onRender(var1, var2, var4, var6, var8, var9)) return;
                 this.renderEntityOnFire(var1, var2, var4, var6, var9);
             }
 
