--- /net/minecraft/item
+++ /net/minecraft/item
@@ -218,7 +218,7 @@
             return false;
         } else {
             if(var1 > 0) {
-                int var3 = EnchantmentHelper.getEnchantmentLevel(Enchantments.UNBREAKING, this);
+                int var3 = EnchantmentHelper.getEnchantmentLevel(Enchantments.UNBREAKING, this, null);
                 int var4 = 0;
 
                 for(int var5 = 0; var3 > 0 && var5 < var1; ++var5) {
