--- /net/minecraft/client/gui
+++ /net/minecraft/client/gui
@@ -345,7 +345,7 @@
     }
 
     protected void renderPotionEffects(ScaledResolution var1) {
-        Collection var2 = this.mc.thePlayer.getActivePotionEffects();
+        Collection<PotionEffect> var2 = this.mc.thePlayer.getActivePotionEffects();
         if(!var2.isEmpty()) {
             this.mc.getTextureManager().bindTexture(GuiContainer.INVENTORY_BACKGROUND);
             GlStateManager.enableBlend();
@@ -552,7 +552,7 @@
                 return var1.getPlayerName() != null && !var1.getPlayerName().startsWith("#");
             }
         }));
-        ArrayList var21;
+        ArrayList<Score> var21;
         if(var5.size() > 15) {
             var21 = Lists.newArrayList(Iterables.skip(var5, var4.size() - 15));
         } else {
